[Bug]: inpaint_only crash with non 512x512 image resolution

### Is there an existing issue for this?

- [X] I have searched the existing issues and checked the recent builds/commits of both this extension and the webui

### What happened?

Since my last update (A1111 v1.8 + ControlNet 1.1.441), I have an error when I want to use the "inpaint_only" processor with an image width different of 512.

### Steps to reproduce the problem

Start A1111
Go to txt2img
Enable ControlNet
Set "inpaint only" processor
Load an image in ControlNet
**Set image Width != 512**
Click "generate"

### What should have happened?

On the console, I get the following error : 

```
2024-03-04 18:54:20,508 - ControlNet - INFO - unit_separate = False, style_align = False██████| 20/20 [00:03<00:00,  6.14it/s]
2024-03-04 18:54:20,511 - ControlNet - INFO - Loading model from cache: control_v11p_sd15_inpaint [ebff9138]
2024-03-04 18:54:20,511 - ControlNet - WARNING - Warn: Using legacy field 'p.controlnet_input_image'
2024-03-04 18:54:20,614 - ControlNet - INFO - Using preprocessor: inpaint_only
2024-03-04 18:54:20,615 - ControlNet - INFO - preprocessor resolution = -1
*** Error running process: C:\Users\Jyce\Desktop\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py
    Traceback (most recent call last):
      File "C:\Users\Jyce\Desktop\stable-diffusion-webui\modules\scripts.py", line 784, in process
        script.process(p, *script_args)
      File "C:\Users\Jyce\Desktop\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 1279, in process
        self.controlnet_hack(p)
      File "C:\Users\Jyce\Desktop\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 1264, in controlnet_hack
        self.controlnet_main_entry(p)
      File "C:\Users\Jyce\Desktop\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 1144, in controlnet_main_entry
        final_inpaint_mask = cv2.dilate(final_inpaint_mask, np.ones((sigma, sigma), dtype=np.uint8))
    cv2.error: OpenCV(4.9.0) D:\a\opencv-python\opencv-python\opencv\modules\core\src\matrix.cpp:1099: error: (-215:Assertion failed) dims <= 2 && step[0] > 0 in function 'cv::Mat::locateROI'
```

### Commit where the problem happens

https://github.com/Mikubill/sd-webui-controlnet/commit/461f8c5f169f5b69971e846279479b050d330744

### What browsers do you use to access the UI ?

Google Chrome

### Command Line Arguments

```Shell
set COMMANDLINE_ARGS=--xformers --opt-channelslast --no-half-vae --api
```


### List of enabled extensions

sd-webui-controlnet
sd-webui-segment-anything

### Console logs

```Shell
venv "C:\Users\Jyce\Desktop\stable-diffusion-webui\venv\Scripts\Python.exe"
Python 3.10.6 (tags/v3.10.6:9c7b4bd, Aug  1 2022, 21:53:49) [MSC v.1932 64 bit (AMD64)]
Version: v1.8.0
Commit hash: bef51aed032c0aaa5cfd80445bc4cf0d85b408b5
Launching Web UI with arguments: --xformers --opt-channelslast --no-half-vae --api --listen --device-id 0 --port 7860
[-] ADetailer initialized. version: 24.3.0, num models: 10
ControlNet preprocessor location: C:\Users\Jyce\Desktop\stable-diffusion-webui\extensions\sd-webui-controlnet\annotator\downloads
2024-03-04 19:09:38,211 - ControlNet - INFO - ControlNet v1.1.441
2024-03-04 19:09:38,289 - ControlNet - INFO - ControlNet v1.1.441
Loading weights [f93e6a50ac] from C:\Users\Jyce\Desktop\stable-diffusion-webui\models\Stable-diffusion\uberRealisticPornMerge_urpmv13.safetensors
2024-03-04 19:09:38,825 - ControlNet - INFO - ControlNet UI callback registered.
Creating model from config: C:\Users\Jyce\Desktop\stable-diffusion-webui\configs\v1-inference.yaml
Running on local URL:  http://0.0.0.0:7860
Applying attention optimization: xformers... done.
Model loaded in 2.5s (load weights from disk: 0.7s, create model: 0.3s, apply weights to model: 1.3s, calculate empty prompt: 0.1s).

To create a public link, set `share=True` in `launch()`.
Startup time: 15.5s (prepare environment: 2.5s, import torch: 3.4s, import gradio: 0.7s, setup paths: 0.8s, initialize shared: 0.2s, other imports: 0.4s, load scripts: 2.3s, create ui: 0.5s, gradio launch: 4.4s, add APIs: 0.4s).
2024-03-04 19:09:53,044 - ControlNet - INFO - unit_separate = False, style_align = False
2024-03-04 19:09:53,197 - ControlNet - INFO - Loading model: control_v11p_sd15_inpaint [ebff9138]
2024-03-04 19:09:53,710 - ControlNet - INFO - Loaded state_dict from [C:\Users\Jyce\Desktop\stable-diffusion-webui\models\ControlNet\control_v11p_sd15_inpaint.pth]
2024-03-04 19:09:53,710 - ControlNet - INFO - controlnet_default_config
2024-03-04 19:09:55,122 - ControlNet - INFO - ControlNet model control_v11p_sd15_inpaint [ebff9138](ControlModelType.ControlNet) loaded.
2024-03-04 19:09:55,123 - ControlNet - WARNING - Warn: Using legacy field 'p.controlnet_input_image'
2024-03-04 19:09:55,228 - ControlNet - INFO - Using preprocessor: inpaint_only
2024-03-04 19:09:55,228 - ControlNet - INFO - preprocessor resolution = -1
*** Error running process: C:\Users\Jyce\Desktop\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py
    Traceback (most recent call last):
      File "C:\Users\Jyce\Desktop\stable-diffusion-webui\modules\scripts.py", line 784, in process
        script.process(p, *script_args)
      File "C:\Users\Jyce\Desktop\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 1279, in process
        self.controlnet_hack(p)
      File "C:\Users\Jyce\Desktop\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 1264, in controlnet_hack
        self.controlnet_main_entry(p)
      File "C:\Users\Jyce\Desktop\stable-diffusion-webui\extensions\sd-webui-controlnet\scripts\controlnet.py", line 1144, in controlnet_main_entry
        final_inpaint_mask = cv2.dilate(final_inpaint_mask, np.ones((sigma, sigma), dtype=np.uint8))
    cv2.error: OpenCV(4.9.0) D:\a\opencv-python\opencv-python\opencv\modules\core\src\matrix.cpp:1099: error: (-215:Assertion failed) dims <= 2 && step[0] > 0 in function 'cv::Mat::locateROI'


---
100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [00:03<00:00,  5.51it/s]
Total progress: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [00:03<00:00,  5.63it/s]
Total progress: 100%|███████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████████| 20/20 [00:03<00:00,  6.29it/s]
```


### Additional information

_No response_

--- a/scripts/controlnet.py
+++ b/scripts/controlnet.py
@@ -1137,8 +1137,12 @@ class Script(scripts.Script, metaclass=(
                 final_inpaint_mask = final_inpaint_feed[:, 3, :, :].astype(np.float32)
                 final_inpaint_raw = final_inpaint_feed[:, :3].astype(np.float32)
                 sigma = shared.opts.data.get("control_net_inpaint_blur_sigma", 7)
-                final_inpaint_mask = cv2.dilate(final_inpaint_mask, np.ones((sigma, sigma), dtype=np.uint8))
-                final_inpaint_mask = cv2.blur(final_inpaint_mask, (sigma, sigma))[:, None]
+                final_inpaint_mask_post_cv = []
+                for m in final_inpaint_mask:
+                    m = cv2.dilate(m, np.ones((sigma, sigma), dtype=np.uint8))
+                    m = cv2.blur(m, (sigma, sigma))[None, None]
+                    final_inpaint_mask_post_cv.append(m)
+                final_inpaint_mask = np.concatenate(final_inpaint_mask_post_cv, axis=0)
                 _, _, Hmask, Wmask = final_inpaint_mask.shape
                 final_inpaint_raw = torch.from_numpy(np.ascontiguousarray(final_inpaint_raw).copy())
                 final_inpaint_mask = torch.from_numpy(np.ascontiguousarray(final_inpaint_mask).copy())

